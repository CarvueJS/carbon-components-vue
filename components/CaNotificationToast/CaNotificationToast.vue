<template>
  <div data-notification class="bx--toast-notification" :class="dynamicClass" role="alert">
    <div class="bx--toast-notification__details">
      <h3 class="bx--toast-notification__title">{{title}}</h3>
      <p class="bx--toast-notification__subtitle">{{subtitle}}</p>
      <p class="bx--toast-notification__caption">{{caption}}</p>
    </div>
    <button data-notification-btn class="bx--toast-notification__close-button" type="button">
      <svg class="bx--toast-notification__icon" aria-label="close" width="10" height="10" viewBox="0 0 10 10" fill-rule="evenodd">
        <path d="M9.8 8.6L8.4 10 5 6.4 1.4 10 0 8.6 3.6 5 .1 1.4 1.5 0 5 3.6 8.6 0 10 1.4 6.4 5z"></path>
      </svg>
  </button>
</div>

</template>

<script>
export default {
  name: 'ca-notification-toast',
  props: {
    error: {
      type: Boolean,
      default: false,
    },
    info: {
      type: Boolean,
      default: false,
    },
    success: {
      type: Boolean,
      default: false,
    },
    warning: {
      type: Boolean,
      default: false,
    },
    title: {
      type: String,
      default: '',
    },
    subtitle: {
      type: String,
      default: '',
    },
    caption: {
      type: String,
      default: '',
    },
  },
  computed: {
    dynamicClass() {
      return {
        'bx--toast-notification--error': this.error,
        'bx--toast-notification--info': this.info,
        'bx--toast-notification--success': this.success,
        'bx--toast-notification--warning': this.warning,
      };
    },
  },
};
</script>

<style></style>
