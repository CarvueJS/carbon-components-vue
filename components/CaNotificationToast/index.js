import CaNotificationToast from './CaNotificationToast.vue';

CaNotificationToast.install = (Vue) => {
  Vue.component(CaNotificationToast.name, CaNotificationToast);
};

export default CaNotificationToast;
