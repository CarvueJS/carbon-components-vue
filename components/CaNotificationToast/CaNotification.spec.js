import { shallow } from 'vue-test-utils';
import CaNotificationToast from './CaNotificationToast.vue';

describe('CaNotificationToast', () => {
  test('is a Vue instance', () => {
    const wrapper = shallow(CaNotificationToast);
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  test('Mathes snapshot', () => {
    const wrapper = shallow(CaNotificationToast);
    expect(wrapper.html()).toMatchSnapshot()
  });
})
