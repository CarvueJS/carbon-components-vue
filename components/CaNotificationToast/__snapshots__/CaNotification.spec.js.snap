// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CaNotificationToast Mathes snapshot 1`] = `
<div data-notification="" role="alert" class="bx--toast-notification">
  <div class="bx--toast-notification__details">
    <h3 class="bx--toast-notification__title"></h3>
    <p class="bx--toast-notification__subtitle"></p>
    <p class="bx--toast-notification__caption"></p>
  </div>
  <button data-notification-btn="" type="button" class="bx--toast-notification__close-button">
    <svg aria-label="close" width="10" height="10" viewBox="0 0 10 10" fill-rule="evenodd" class="bx--toast-notification__icon">
      <path d="M9.8 8.6L8.4 10 5 6.4 1.4 10 0 8.6 3.6 5 .1 1.4 1.5 0 5 3.6 8.6 0 10 1.4 6.4 5z"></path>
    </svg>
  </button>
</div>
`;
